diff --git a/arch/arm/boot/dts/bcm2710-rpi-3-b-plus.dts b/arch/arm/boot/dts/bcm2710-rpi-3-b-plus.dts
index 55420ac94..77774868b 100644
--- a/arch/arm/boot/dts/bcm2710-rpi-3-b-plus.dts
+++ b/arch/arm/boot/dts/bcm2710-rpi-3-b-plus.dts
@@ -10,7 +10,7 @@
 	model = "Raspberry Pi 3 Model B+";
 
 	chosen {
-		bootargs = "coherent_pool=1M 8250.nr_uarts=1";
+		bootargs = "coherent_pool=1M 8250.nr_uarts=1 console=ttyAMA0,115200 console=tty1 root=/dev/mmcblk0p7 rootfstype=ext4 elevator=deadline fsck.repair=yes rootwait splash plymouth.ignore-serial-consoles";
 	};
 
 	aliases {
@@ -18,6 +18,10 @@
 		serial1 = &uart0;
 		mmc1 = &mmcnr;
 	};
+	memory@0 {
+		reg = <0x00 0x3c000000>;
+		device_type = "memory";
+	};
 };
 
 &gpio {
diff --git a/arch/arm/boot/dts/bcm2710-rpi-3-b.dts b/arch/arm/boot/dts/bcm2710-rpi-3-b.dts
index 261827cdb..15bbd93a6 100644
--- a/arch/arm/boot/dts/bcm2710-rpi-3-b.dts
+++ b/arch/arm/boot/dts/bcm2710-rpi-3-b.dts
@@ -10,7 +10,7 @@
 	model = "Raspberry Pi 3 Model B";
 
 	chosen {
-		bootargs = "coherent_pool=1M 8250.nr_uarts=1";
+		bootargs = "coherent_pool=1M 8250.nr_uarts=1 earlyprintk loglevel=8 console=ttyAMA0,115200 console=tty1 root=/dev/mmcblk0p2 rw rootwait";
 	};
 
 	aliases {
@@ -18,6 +18,10 @@
 		serial1 = &uart0;
 		mmc1 = &mmcnr;
 	};
+	memory@0 {
+		reg = <0x00 0x3c000000>;
+		device_type = "memory";
+	};
 };
 
 &gpio {
