diff --color -ruN B/FspUpd.h A/FspUpd.h
--- B/FspUpd.h	2023-04-17 12:40:25.611566326 +0000
+++ A/FspUpd.h	2023-04-17 12:35:32.811108287 +0000
@@ -33,8 +33,6 @@
 #ifndef __FSPUPD_H__
 #define __FSPUPD_H__
 
-#include <FspEas.h>
-
 #pragma pack(1)
 
 #define FSPT_UPD_SIGNATURE               0x545F4450554D4551        /* 'QEMUPD_T' */
diff --color -ruN B/FspmUpd.h A/FspmUpd.h
--- B/FspmUpd.h	2023-04-17 12:40:25.614899665 +0000
+++ A/FspmUpd.h	2023-04-17 12:36:05.284489504 +0000
@@ -33,7 +33,9 @@
 #ifndef __FSPMUPD_H__
 #define __FSPMUPD_H__
 
-#include <FspUpd.h>
+#include "stdtype_mapping.h"
+#include "FspCommon.h"
+#include "FspUpd.h"
 
 #pragma pack(1)
 
